tests/test_doxygen_output.py

```diff
@@ -1,51 +1,32 @@
-import sys
-import unicodedata
 import re
-from glob import *
+import glob
 import unittest
 
+DOC_FILES = glob.glob("doc/html/*.htm*")
 
 class TestDoxygenOutput(unittest.TestCase):
     """ Test class for the doxygen output. """
 
     def test_hash(self):
         ''' Test case is checking if there are illegal hash chars in the documentation. -> doxygen link not found.  '''
-        for file in glob("doc/html/*.htm*"):
-            with open(file, "rt") as fin:
-                i = 0
-
-                for line in fin:
-                    i += 1                    
-                    matchHash = re.search(r"([\s>]|^)#\w(\S)*", line)
-
-                    if matchHash is not None:
-                        self.assertIsNone(matchHash, file + " in line " + str(i) + ": not permitted hash found. Search for: '"+ line[matchHash.start():matchHash.end()])
+        for file in DOC_FILES:
+            with open(file, "rt") as fin, self.subTest(file=file):
+                for i, line in enumerate(fin, start=1):
+                    self.assertNotRegex(line, r"([\s>]|^)#\w(\S)*", file + " in line " + str(i) + ": not permitted hash found.")
```

Comment:
```
FAIL: test_hash (test_doxygen_output.TestDoxygenOutput) (file='doc/html/windows.html')
Test case is checking if there are illegal hash chars in the documentation. -> doxygen link not found.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "open-simulation-interface/tests/test_doxygen_output.py", line 17, in test_hash
    self.assertNotRegex(line, r"([\s>]|^)#\w(\S)*", file + " in line " + str(i) + ": not permitted hash found.")
AssertionError: Regex matched: '#lol' matches '([\\s>]|^)#\\w(\\S)*' in '#lol\n' : doc/html/windows.html in line 5: not permitted hash found.
```

I think the assertion message is good enough so that we dont need to manually add the offending section again.

Reply:
You may be right!

Reply:
Good point.